subprojects/packagefiles/speexdsp/meson.build

```diff
@@ -0,0 +1,113 @@
+project(
+  'speexdsp',
+  'c',
+  version : '1.2.0',
+  license : 'bsd-like',
+  default_options : [
+    'c_std=c99',
+    'buildtype=release',
+    'default_library=both',
+    'b_lto=true',
+    'warning_level=0'])
```

Comment:
SpeexDSP is performance-critical. With DOSBox Staging, even when optimized it impacts our global FPS by ~10%.

I want to provide the most informed build options out-of-the-box against the most probable use case, with the theory that it should be easy to get a "good result" and hard to get a bad result.

This means:
 - Providing both static and shared libraries (maximizes flexibility out-of-the-box)
 - Providing an optimized build (Meson gives us LTO too and speexDSP works fine with it)
 - Eliminating visual clutter (reducing warnings: they're still there.. and if someone wants to dig in and help a project, they can turn them on and fix them).

Reply:
Why is it flexible to provide both static and shared? When linking to it as a subproject, it will only ever use the shared version anyway. And in practice, if you build vendored wraps as a subproject, you want to link statically so that the built program is more relocatable.

As far as lto goes, I totally agree lto is great and you should use it, but... as a per-subproject setting it's entirely useless, because Meson ignores it.

Instead, you should be setting lto in the superproject (e.g. dosbox) and it will apply to the entire build, including subprojects.

As far as warnings go, I'm indifferent to them :) but note that you can always turn them off with `dependency('foobar', default_options: ['warning_level=0'])` even if you're using a wrap someone else contributed which has it set higher.

Reply:
> Why is it flexible to provide both static and shared? 

If there's anyone using the meson files as-is (not as vendored in wraps) or if there are repo packagers using the meson build files, then this would generate both static and shared libraries, which is often lacking (and missed) by packagers. Being able to satisfy both shared linking and static lib requests ==  more flexible. But perhaps this usecase isn't common.

> When linking to it as a subproject, it will only ever use the shared version anyway. And in practice, if you build vendored wraps as a subproject, you want to link statically so that the built program is more relocatable.

Indeed; for our release builds (and when giving users build options), we already turn on all the static-lib options (they're quite a mess of them).

Maybe it makes sense for me to default to a static library -- what do you suggest?

> As far as lto goes, I totally agree lto is great and you should use it, but... as a per-subproject setting it's entirely useless, because Meson ignores it. 

Yes, we use LTO explicitly; so this setting is useless and will never be active in any "default" build scenario?  In that case.. I'll knock it out - but will hold off for you to confirm.

> Instead, you should be setting lto in the superproject (e.g. dosbox) and it will apply to the entire build, including subprojects.

We do set it explicitly on the command line, but not by default at the project level - because those builds are expensive and can't be ccache'd.  But LTO is cheap for speexDSP, so I'll explicitly set it for that in our defaults.

> As far as warnings go, I'm indifferent to them :) but note that you can always turn them off with dependency('foobar', default_options: ['warning_level=0']) even if you're using a wrap someone else contributed which has it set higher.

:+1: , good to know - and I will add that for our other wraps (which also flood the console).